processor/deltatocumulativeprocessor/internal/streams/expiry.go

```diff
@@ -0,0 +1,68 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package streams // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/deltatocumulativeprocessor/internal/streams"
+
+import (
+	"context"
+	"time"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/exp/metrics/identity"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/exp/metrics/staleness"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/exp/metrics/streams"
+)
+
+func ExpireAfter[T any](ctx context.Context, items streams.Map[T], ttl time.Duration) streams.Map[T] {
```

Comment:
IMO, wrapping the expiry functionality in a struct obfuscates how things work together. And makes it harder to do things like mutex sharing. You're wrapping some functions, but then exposing other base functions.

It would be clearer to just have a plain method in say, Processor() which did this behavior, and took in pointers